umbraco/Umbraco-CMS
# Netcore: Migreate macro related controllers

### Notes
- Migrated controllers
    - `MacrosController`
    - `MacroRenderingController`
- Fixed issues and migrated 
  - Stuff to render templates and macros to strings
- Fixed issue with filesystems rooted a `~/wwwroot` instead of `~/`
- Activated runtime view compilation.
- Prepared the new `BackOfficeServerVariables` to generate links using `LinkGenerator` and added new endpoints
- `PublicAccessChecker` still needs to be implemented. This PR contains a pass through version.

### Tests
The test is pretty complicated at the moment, due to authentication etc.
So you can see the recording from the last demo session, as a proof, it is working :)

I started by making the following changes:

In `MacrosController.cs`
```diff
- [UmbracoTreeAuthorize(Constants.Trees.Macros)]
+// [UmbracoTreeAuthorize(Constants.Trees.Macros)]
```
* In `UmbracoAuthorizedApiController.cs`
```diff
-  [UmbracoAuthorize]
+ //[UmbracoAuthorize]
```
* In `UmbracoAuthorizedJsonController.cs`
```diff
-  [TypeFilter(typeof(ValidateAngularAntiForgeryTokenAttribute))]
+ //[TypeFilter(typeof(ValidateAngularAntiForgeryTokenAttribute))]
```
After that, I tested and used the `MacrosController` and `CodeFileController` to create data to test `MacroRenderingController`. Futhermore I used v8, to create a database with a content of Id `11472` and I copied over a view called `~/Views/Root.cshtml` used by that content type.

1. Create  a new macro
   * POST - https://localhost:44331/umbraco/backoffice/UmbracoApi/Macros/Create?name=netcore
   * Save the id returned. In my case `6063`
1. Get the newly created macro
   * GET - https://localhost:44331/umbraco/backoffice/UmbracoApi/Macros/GetById?id=6063 (Use your own id)
   * Save the returned json to use and modify later
1. Save a partial macro file
    * POST - https://localhost:44331/umbraco/backoffice/UmbracoApi/CodeFile/PostSave
    * `{"virtualPath":"/Views/MacroPartials/","path":null,"name":"netcore","content":"@inherits Umbraco.Web.Common.Macros.PartialViewMacroPage\r\n@inject Umbraco.Core.Net.IUmbracoComponentRenderer UmbracoComponentRenderer\r\n<b>Test</b>\r\n@UmbracoComponentRenderer.RenderTemplate(11472)","fileType":"partialViewMacros","snippet":null,"id":"0","notifications":[]}`
1. Update the macro to use the new view
    * POST - https://localhost:44331/umbraco/backoffice/UmbracoApi/Macros/Save 
    * `{
"notifications": [],
"useInEditor": false,
"renderInEditor": true,
"cachePeriod": 0,
"cacheByPage": false,
"cacheByUser": false,
"view": "~/Views/MacroPartials/netcore.cshtml",
"parameters": [],
"name": "netcore",
"id": 6063,
"udi": "umb://macro/15f9d367253144f2b387cfe8ab4f73d1",
"icon": "icon-settings-alt",
"trashed": false,
"key": "15f9d367-2531-44f2-b387-cfe8ab4f73d1",
"parentId": -1,
"alias": "netcore",
"path": "-1,6063",
"metaData": {}
}`
1. Get the HTML result from a macro
    * GET - https://localhost:44331/umbraco/backoffice/UmbracoApi/MacroRendering/GetMacroResultAsHtmlForEditor?macroAlias=netcore&pageId=11472 (Use your own id)
